charts/sourcegraph/templates/cadvisor/cadvisor.DaemonSet.yaml

```diff
@@ -82,6 +82,11 @@ spec:
         - name: disk
           mountPath: /dev/disk
           readOnly: true
+        {{- if .Values.cadvisor.containerSecurityContext.privileged }}
+        - name: kmsg
+          mountPath: /dev/kmsg
+          readOnly: true
+        {{- end }}
```

Comment:
maybe I am overcomplicating things.

without `privileged` and we mount the `/dev/kmsg`, users would see the below logs from `cadvisor`.

```
k logs daemonset/cadvisor                                                     14:05:31
Found 2 pods, using pod/cadvisor-qkdlk
W0505 21:05:23.014872       1 machine_libipmctl.go:63] There are no NVM devices!
W0505 21:05:23.068489       1 manager.go:296] Could not configure a source for OOM detection, disabling OOM events: open /dev/kmsg: operation not permitted
``

If we don't mount `/dev/kmsg`, users will always see something like

```
 Could not configure a source for OOM detection, disabling OOM events: open /dev/kmsg: no such file or directory
```

either one of them is not ideal, so I figure we should just don't mount unnecessary stuff when it has no used (`privielged=false`)